### PR TITLE
[GLUTEN-2858][VL][CI] Use pre-build docker image for CI

### PR DESCRIPTION
**Pull request by xieqi**
## What changes were proposed in this pull request?

Use pre-build docker image for CI

## How was this patch tested?

Use exist UTs

### CODE REVIEW

**Comment by zhztheplayer** [comment]
Would it be better to maintain a Dockerfile in our repo for letting users build the image? So one could reproduce the CI error within duplicated container in his/her local environment.

File: .github/workflows/velox_be.yml [hunk 1]
```diff
@@ -44,33 +44,17 @@ concurrency:
 
 jobs:
   velox-ubuntu2004-test-spark32:
-    runs-on: velox-self-hosted
+    runs-on: velox-self-hosted-image
```
Comment (winningsix) [inline .github/workflows/velox_be.yml:47]: Question: what's the difference between ```velox-self-hosted-image``` and ```velox-self-hosted```? Is it on K8S?

File: .github/workflows/velox_be.yml [hunk 2]
```diff
@@ -44,33 +44,17 @@ concurrency:
...
-          cd /opt/gluten/ep/build-velox/src && \
-          ./get_velox.sh && \
-          ./build_velox.sh --build_benchmarks=ON'
+          docker run --rm --init --privileged --ulimit nofile=65536:65536 --ulimit core=-1 --security-opt seccomp=unconfined -v $PWD:/opt/gluten --name velox-ubuntu2004-test-$GITHUB_RUN_ID -e NUM_THREADS=30 -dit 10.0.2.4:5000/gluten-dev/ubuntu:20.04
       - name: Build Gluten CPP library
         run: |
           docker exec velox-ubuntu2004-test-$GITHUB_RUN_ID bash -c '
           cd /opt/gluten/cpp && \
-          mkdir build && cd build && cmake -DBUILD_TESTS=ON -DBUILD_EXAMPLES=ON -DBUILD_BENCHMARKS=ON .. && make -j'
+          ./compile.sh --build_velox_backend=ON --arrow_home=/opt/arrow --velox_home=/opt/velox --build_tests=ON --build_benchmarks=ON --enable_s3=ON  --enable_hdfs=ON'
```
Comment (winningsix) [inline .github/workflows/velox_be.yml:57]: Question: Why do we want to switch to ```compile.script```? Previously, Velox build and Gluten CPP build are separated jobs.

File: .github/workflows/velox_be.yml [hunk 3]
```diff
@@ -237,246 +161,3 @@ jobs:
         if: ${{ always() }}
         run: |
           docker stop velox-ubuntu2004-test-spark33-$GITHUB_RUN_ID || true
-         
-  velox-ubuntu2204-test:
```
Comment (winningsix) [inline .github/workflows/velox_be.yml:241]: Delete by mistake?

**Comment by zhouyuan** [comment]
@xieqi thanks for improving, can you please list which parts are pre-installed in the docker image?

**Comment by xieqi** [comment]
> @xieqi thanks for improving, can you please list which parts are pre-installed in the docker image?

spark322、spark331、ccache、maven cache、vcpkg cache、pre-build arrow and velox

File: .github/workflows/velox_be.yml [hunk 4]
```diff
@@ -195,77 +147,43 @@ jobs:
...
-          OS_IMAGE=ubuntu:20.04 \
-          tools/gluten-te/ubuntu/cbash.sh sleep 14400
-      - name: Setup maven cache
-        run: |
-          docker cp ~/.m2/repository velox-ubuntu2004-test-spark33-$GITHUB_RUN_ID:/root/.m2/
-      - name: Compile Spark3.3
-        run: |
-          docker exec velox-ubuntu2004-test-spark33-$GITHUB_RUN_ID bash -c '
-          cd /var/cache/spark33 && \
-          cp /opt/gluten/dev/get_spark_env.sh . &&
```
Comment (PHILO-HE) [inline .github/workflows/velox_be.yml:215]: The script `get_spark_env.sh` will be useless. Let's remove it. Thanks!

File: .github/workflows/velox_be.yml (hunk 1 above)
Comment (xieqi) [reply .github/workflows/velox_be.yml]: not in K8S, velox-self-hosted-image can pull image from private docker registry

**Comment by PHILO-HE** [comment]
> Would it be better to maintain a Dockerfile in our repo for letting users build the image? So one could reproduce the CI error within duplicated container in local environment.

Agree! @xieqi, could you please add the Dockerfile into the repo? It is quite helpful. Thanks!

**Comment by winningsix** [comment]
> cpkg ca

We should add those to code comments. 
```
spark322、spark331、ccache、maven cache、vcpkg cache、pre-build arrow and velox、celeborn-0.3.0-incubating-bin
```

**Comment by winningsix** [comment]
Let's on hold this PR merge to be used as validation. Once image pre-build image was online, we will use this to test and merge it later.

**Review by PHILO-HE — APPROVED** [review]

**Comment by zhztheplayer** [comment]
> We can simplify the commands in building external project & gluten native code. I will do some follow-up work.
> 
> @zhztheplayer, please let me know if you have other comment. Thanks!

Thanks!

**Pull request merged**